.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: NAME-EXAMINATION UI CI
+
+on:
+  pull_request:
+    types: [assigned, synchronize]
+    paths:
+        - "web/**"
+
+defaults:
+  run:
+    shell: bash
+    working-directory: ./web
+
+jobs:
+  setup-job:
+    runs-on: ubuntu-20.04
+
+    if: github.repository == 'bcgov/name-examination'
+
+    steps:
+      - uses: actions/checkout@v3
+      - run: "true"
+
+  linting:
+    needs: setup-job
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        node-version: [18.16.0]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: |
+          npm ci
+      - name: Linting
+        run: |
+          npm run lint
+
+  testing-coverage:
+    needs: setup-job
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        node-version: [18.16.0]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: |
+          npm ci
+      - name: Test with Jest
+        id: test
+        run: |
+          npm run test
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+            flags: analyticsui
```

Comment:
name need to be change. And need to create codecov.yaml into the project as well.